lib/ansible/modules/system/nfsexport.py

```diff
@@ -0,0 +1,204 @@
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+
+__docformat__ = 'reStructuredText'
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'supported_by': 'community',
+    'status': ['preview']
+}
+
+DOCUMENTATION = """
+---
+module: nfsexport
+version_added: "2.4"
+short_description: This module will configure a nfs exports file
+description:
+    - This module will configure an nfs share by writing an entry in the exports file.
+notes:
+    - This module only changes the exports file itself. If changes are made, the
+      exportfs command should be run and the nfs daemon reloaded.
+author: "Randy Nance (randynobx) <randynobx@gmail.com>"
+options:
+    path:
+        description:
+            - Directory to export.
+        required: true
+    dest:
+        description:
+           - Export file to write to.
+        default: /etc/exports
+    clients:
+        description:
+            - list of nfs clients and host specific options.
+        default: "*"
+    options:
+        description:
+            - Nfs export options for all clients in export entry.
+    state:
+        description:
+            - Whether to add or remove export path from file.
+        default: present
+        choices:
+            - present
+            - absent
+"""
+
+EXAMPLES = """
+- name: ensure "/home 10.0.0.1(rw)" exists in /etc/exports
+  nfsexport:
+    path: /home
+    clients: 10.0.0.1(rw)
+    state: present
+
+- name: ensure "/nfs/share -sync,root_squash 10.0.0.1(rw) 10.1.0.0/24(ro)" exists in /etc/exports
+  nfsexport:
+    path: /nfs/share
+    clients:
+      - 10.0.0.1(rw)
+      - 10.1.0.0/24(ro)
+    options: sync,root_squash
+
+- name: remove "/home/user *" from /etc/nfsexports if present
+  nfsexport:
+    path: /home/user
+    dest: /etc/nfsexports
+    state: absent
+
+- name: ensure "/nfs/share *(ro,sync)" exists in /etc/exports
+  nfsexport:
+    path: /nfs/share
+    options: ro,sync
+"""
+
+RETURN = """
+
+"""
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+def write_to_file(mod, filepath, mode, content):
+    try:
+        fd = open(filepath, mode)
+    except IOError as err:
+        mod.fail_json(msg="failed to open %s for writing: %s" % (filepath, str(err)))
+    fd.writelines(content)
+    fd.close()
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            path=dict(required=True, type='path'),
+            dest=dict(required=False, default='/etc/exports'),
+            clients=dict(required=False, default='*'),
+            options=dict(required=False, default=''),
+            state=dict(required=False, default='present', choices=['absent', 'present']),
+        ),
+        supports_check_mode=True
+    )
+
+    params = module.params
+    path = params['path'].strip()
+    dest = params['dest'].strip()
+    state = params['state']
+
+    # build list of clients
+    client_list = params['clients'] \
```

Comment:
IMHO this statement is hard to read. Woul rewrite to simple if else and for loop.